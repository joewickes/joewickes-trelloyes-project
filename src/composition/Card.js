import React from 'react';
import './Card.css';

class CardClass extends React.Component {
  
  render() {
    const {title, content, deleteCard, id} = this.props;

    return (
      <div className="Card">
        <button type="button" id={id} onClick={(e) => deleteCard(e.target.id)}>delete</button>
        <h3>{title}</h3>
        <p>{content}</p>
      </div>
    );
  }
}

export default CardClass;