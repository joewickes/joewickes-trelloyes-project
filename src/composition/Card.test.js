import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Card from './Card';

describe('Card Component', () => {

  // The component renders without crashing
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Card />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // given the 
  it('renders the Card as expected', () => {
    const wrapper = shallow(<Card title="Title" content="Content"/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });


  // when remove card is clicked, it removes card from state (lists and allCards)
  it('removes the right card when deleted', function() {
    const wrapper = shallow(<Card title="Title" content="Content"/>);
    wrapper.find('button').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});