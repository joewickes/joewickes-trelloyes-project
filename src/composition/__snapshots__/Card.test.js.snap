// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders the Card as expected 1`] = `
<div
  className="Card"
>
  <button
    onClick={[Function]}
    type="button"
  >
    delete
  </button>
  <h3>
    Title
  </h3>
  <p>
    Content
  </p>
</div>
`;
